File: online judge.kaner murillo/python/black list/11321.py
#https://onlinejudge.org/external/113/11321.pdf
#11321

from sys import stdin

cantidad, mod = [int(x) for x in stdin.readline().split()]

def comparar(a,b):

    ans = False

    if a%mod == b%mod:
        
        if a%2 == 0 and b%2 == 0:

            ans = a<b
        
        elif a%2 == 0 and b%2!= 0:

            ans = False
        
        elif a%2!=0 and b%2 == 0:
            
            ans = True

        else:
            ans = a > b

    else:

        ans = a>b

    return ans

while cantidad != 0 or mod != 0:

    lista = []

    for i in range(cantidad):

        lista.append(int(stdin.readline().strip()))

    lista.sort(comparar)

    print(cantidad, mod)

    for i in lista:
        print(i)
        
    cantidad, mod = [int(x) for x in stdin.readline().split()]

print(0,0)

File: online judge.kaner murillo/c++/11321.cpp
#include <algorithm>
#include <cstdio>
#include <iostream>
#include <vector>

using namespace std;

int mod;

bool comparar(int a, int b){

    bool ans;

    if(a%mod == b%mod){

        if(a%2 != 0 && b%2 !=0){
            ans = a>b;
        }
    
        else if(a%2 != 0 && b%2 == 0){
            ans = true;
        }

        else if(a%2 == 0 && b%2 != 0){
            ans = false;
        }

        else{
            ans = a<b;
        }
    }

    else{

        ans = a%mod < b%mod;
    }
    

    return ans;
}

int main(){

    int cantidad, i, v;
    vector<int>numeros;

    while(scanf("%i %i", &cantidad, &mod), cantidad != 0 || mod != 0){

        numeros.clear();
        
        for(i = 0; i<cantidad; i++){

            scanf("%i", &v);
            numeros.push_back(v);
        }

        sort(numeros.begin(), numeros.end(), comparar);

        printf("%i %i\n", cantidad, mod);

        for(i = 0; i<numeros.size(); i++){
            printf("%i\n", numeros[i]);
        }
        
    }

    printf("0 0\n");

    return 0;
}
